app/src/main/kotlin/com/wire/android/ui/userprofile/self/SelfUserProfileViewModel.kt

```diff
@@ -78,18 +79,26 @@ class SelfUserProfileViewModel @Inject constructor(
 
     private fun observeEstablishedCall() {
         viewModelScope.launch {
-            observeEstablishedCalls().map { it.isNotEmpty() }.distinctUntilChanged().collect {
-                userProfileState = userProfileState.copy(isUserInCall = it)
-            }
+            observeEstablishedCalls()
```

Comment:
i think it might be 
```
        viewModelScope.launch(dispatchers.io()) {
            observeEstablishedCalls()
                .map { it.isNotEmpty() }
                .distinctUntilChanged()
                .collect {
                    userProfileState = userProfileState.copy(isUserInCall = it)
                }
        }
```
move dispatcher into `launch` and no need to "share" the flow if you are not going to use it more then once

Reply:
Wouldn't that be problematic as we would be updating some UI stuff like `userProfileState` with the `io` thread?